unit-tests/algo/depth-to-rgb-calibration/compare-scene.h

```diff
@@ -48,45 +49,75 @@ void compare_scene( std::string const & scene_dir )
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_x[1], scene_dir, "localRegion_x", 1, 105794, "double_01" ), 105794, 1, compare_same_vectors ));
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_x[2], scene_dir, "localRegion_x", 1, 105794, "double_02" ), 105794, 1, compare_same_vectors ));
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_x[3], scene_dir, "localRegion_x", 1, 105794, "double_03" ), 105794, 1, compare_same_vectors ));
-
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_y[0], scene_dir, "localRegion_y", 1, 105794, "double_00" ), 105794, 1, compare_same_vectors ));
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_y[1], scene_dir, "localRegion_y", 1, 105794, "double_01" ), 105794, 1, compare_same_vectors ));
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_y[2], scene_dir, "localRegion_y", 1, 105794, "double_02" ), 105794, 1, compare_same_vectors ));
-    CHECK( compare_to_bin_file< double >( ir_data.local_region_y[3], scene_dir, "localRegion_y", 1, 105794, "double_03" ), 105794, 1, compare_same_vectors ));
-
-    CHECK( compare_to_bin_file< double >( ir_data.local_edges, scene_dir, "localEdges", 4, 105794, "double_00" ), 105794, 4, compare_same_vectors ));
-    CHECK( compare_to_bin_file< uint8_t >( ir_data.is_supressed, scene_dir, "isSupressed", 1, 105794, "uint8_00" ), 105794, 1, compare_same_vectors ));
+    CHECK(compare_to_bin_file< double >(depth_data.gradient_x, scene_dir, "Zx", z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(depth_data.gradient_y, scene_dir, "Zy",  z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.gradient_x, scene_dir, "Ix",  z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.gradient_y, scene_dir, "Iy",  z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.edges, scene_dir, "iEdge",  z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(depth_data.edges, scene_dir, "zedge",  z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< uint8_t >(depth_data.section_map_depth, scene_dir, "sectionMapDepth",   z_w, z_h, "uint8_00",compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.valid_edge_pixels_by_ir, scene_dir, "validEdgePixelsByIR",  z_w, z_h, "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.valid_location_rc_x, scene_dir, "gridXValid", 1, 105794,  "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.valid_location_rc_y, scene_dir, "gridYValid", 1, 105794,  "double_00", compare_same_vectors));
+    CHECK(compare_to_bin_file< double >(ir_data.valid_location_rc, scene_dir, "locRC", 2, 105794,  "double_00", compare_same_vectors));
```

Comment:
No magic numbers please -- matlab should communicate the dimensions to C++ somehow